### DISCUSSION TITLE
Website unresponsive after deploying to vercel 🥲

### DISCUSSION THREAD

**Discussion by gunaseelan13** [vercel/next.js#41298] [Help] [answered]
Website works well in localhost but once I deploy to vercel website becomes unresponsive and images are not loading.

webpack
site.webmanifest
buildmanifest.js

all loading 🧐

https://github.com/gunaseelan13/ullur.in-Dev

![J_TPR0Vw](https://user-images.githubusercontent.com/16633639/194810483-c234c7c7-8c91-4c59-9a7b-2014c9ca761c.jpg)
![toBcqBHo (1)](https://user-images.githubusercontent.com/16633639/194810492-f3506bbf-ac1a-4df0-9587-648a5607e13c.jpg)

**Comment by icyJoseph** [comment, marked as answer]
Hi,

What happens if on your machine you do `next build && next start`? for example with npm: `npm run build && npm run start`? Is the site still working as you expect?

**Reply by gunaseelan13** [reply]
yes working perfectly in my machine

**Reply by icyJoseph** [reply]
And in your Vercel dashboard the build step finishes correctly right?

**Reply by icyJoseph** [reply]
Btw, probably good to update that webpack config:

```js
 webpack: (config, { dev, nextRuntime }) => {
    // Replace React with Preact only in client production build
    if (!dev && !nextRuntime) {
      Object.assign(config.resolve.alias, {
        react: 'preact/compat',
        'react-dom/test-utils': 'preact/test-utils',
        'react-dom': 'preact/compat',
      });
    }
    return config;
  },
```

`isServer` is ambiguous because it might also be on the Edge runtime, use `nextRuntime` instead, which is `undefined` in the browser.

**Reply by icyJoseph** [reply]
For some reason, your page eats up 100% CPU, which makes me think that there's an infinite loop/runaway effect, which you should be able to see locally, but yeah, I can't really point to it. 

I can see your site with JS disabled though, and it responds when I use an HTTP client from my terminal. 

Make sure the styled components CSS is correct, like, lines end with `;` and proper `{}` enclosing.

**Reply by icyJoseph** [reply]
In this group of components for example: https://github.com/gunaseelan13/ullur.in-Dev/blob/master/components/ac-service/acservicefullstyle.js#L196

There's a lot of broken CSS, which I'd expect to cause problems:

<img width="490" alt="Screenshot 2022-10-10 at 09 39 44" src="https://user-images.githubusercontent.com/21013447/194818545-1aad663d-f8ab-4505-b690-882e452a4fb2.png">

Notice that CSS rule for `a`, that's not closed after line 240, and then there's an anchor tag `<a>` at line 262. That's gonna break the parser, or at least cause some problems.

**Reply by gunaseelan13** [reply]
Thanks for the help (I'm new to development this means a lot 🙏)

No error shown in vercel build time.

I changed webpack config and CSS issues but still the same issue persist. 

One more thing

In previous successfull builds In vercel, middleware was found in build static folder

![JjCzErcj](https://user-images.githubusercontent.com/16633639/194840717-c54543c0-cb6e-4695-a712-ac0f7d093878.png)

But now its not found 

![lD2aHn4y](https://user-images.githubusercontent.com/16633639/194840881-77da7c3b-f3c6-4817-aef8-87e7bf173e6c.png)

**Reply by icyJoseph** [reply]
Aha, but your app does not have middleware though, perhaps you had it before? Still I don't think that's the problem. 

At least now we can see the page, and not just a blank. I do believe that perhaps we doing infinite recursion or some sort of non-obvious infinite loop but I can't figure out where it is happening just now.

**Reply by gunaseelan13** [reply]
🥲 stuck on this for a few days

**Reply by icyJoseph** [reply]
Was it working at any point? Do we need any special setup to run it locally?

**Reply by gunaseelan13** [reply]
It was working till June(last deployed to production)
While updating, last friday(Oct 5, 2022) Vercel went down system wide from then its happening like this

**Reply by gunaseelan13** [reply]
https://github.com/gunaseelan13/ullur.in-Dev/tree/3c30969e4ab82950ab122aa5305771805743033d

Working code

**Reply by icyJoseph** [reply]
I tried to run a git bisect, but starting with e0e8e1613b96751479f523e87205f82b2d627408, something started to make the build fail, and from there I've got no good reference to know which commit really introduced the unwanted behavior.

**Reply by gunaseelan13** [reply]
Same tried deploying the old one which resulted in more issues (ESLint conflicts)

**Reply by icyJoseph** [reply]
You don't need to deploy though, the UI is locked locally as well.

**Reply by gunaseelan13** [reply]
Actually i Repurposed this repo to create mine(bcz of i18n setup) when i deploy this repo its working perfectly both locally and in production.

https://github.com/arturocr/arturocampos.dev

**Reply by icyJoseph** [reply]
I think you need to fix these files:

```
components/healthcarefullpage/healthcarefullstyle.js
components/jobsfullpage/jobsfullpagestyle.js
components/learnfullpage/learnfullpagestyle.js
components/navbar/navstyle.js
components/servicesfull/servicesfullpagestyle.js
styles/articlestyles.js
```

There's broken colors, usage of `:` instead of `;` to terminate CSS declarations, hanging `}` and `</a>` spread around. Install this [extension](https://marketplace.visualstudio.com/items?itemName=styled-components.vscode-styled-components) to help you see the errors.

As to why the issue is happening my guess is that styled-components, with Preact, is not playing nicely at all. 

Either it is the swc compiler, or some kind of API mismatch between what Styled Components, and Preact. If we remove that configuration override from `next.config.js` (and fix the above), things should start to work, I believe.

I notice that repo you reference uses Tailwind rather. Try by removing Preact first, and see if that fixes the site usability, then we can try to chase what's failing when Styled Components hydrates with Preact. I think this the loop, that's locking up the UI, somehow Preact and Styled-Components never finish to parse the styles.

**Reply by gunaseelan13** [reply]
Removed Preact and it worked 🙏 
Thanks a lot.
